examples/game/breakout.rs

```diff
@@ -5,268 +5,381 @@ use bevy::{
-            // brick
+            // Adding one brick at a time
             commands
                 .spawn_bundle(SpriteBundle {
                     material: brick_material.clone(),
                     sprite: Sprite::new(brick_size),
                     transform: Transform::from_translation(brick_position),
                     ..Default::default()
                 })
-                .insert(Collider::Scorable);
+                .insert(Brick)
+                .insert(Collides);
         }
     }
 }
 
-fn paddle_movement_system(
-    keyboard_input: Res<Input<KeyCode>>,
-    mut query: Query<(&Paddle, &mut Transform)>,
-) {
-    if let Ok((paddle, mut transform)) = query.single_mut() {
-        let mut direction = 0.0;
-        if keyboard_input.pressed(KeyCode::Left) {
-            direction -= 1.0;
-        }
-
-        if keyboard_input.pressed(KeyCode::Right) {
-            direction += 1.0;
-        }
+fn spawn_scoreboard(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands
+        .spawn_bundle(TextBundle {
+            text: Text {
+                sections: vec![
+                    TextSection {
+                        value: "Score: ".to_string(),
+                        style: TextStyle {
+                            font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                            font_size: config::SCORE_FONT_SIZE,
+                            color: config::SCOREBOARD_COLOR,
+                        },
+                    },
+                    TextSection {
+                        value: "".to_string(),
+                        style: TextStyle {
+                            font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                            font_size: config::SCORE_FONT_SIZE,
+                            color: config::SCORE_COLOR,
+                        },
+                    },
+                ],
+                ..Default::default()
+            },
+            style: Style {
+                position_type: PositionType::Absolute,
+                position: Rect {
+                    top: config::SCORE_PADDING,
+                    left: config::SCORE_PADDING,
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .insert(Scoreboard);
+}
 
-        let translation = &mut transform.translation;
-        // move the paddle horizontally
-        translation.x += direction * paddle.speed * TIME_STEP;
-        // bound the paddle within the walls
-        translation.x = translation.x.min(380.0).max(-380.0);
+/// Moves everything with both a Transform and a Velovity accordingly
+fn kinematics(mut query: Query<(&mut Transform, &Velocity)>) {
+    for (mut transform, velocity) in query.iter_mut() {
+        transform.translation.x += velocity.x * config::TIME_STEP;
+        transform.translation.y += velocity.y * config::TIME_STEP;
     }
```

Comment:
We should advocate to use `Query::for_each_mut` since it's more performant and should usually be the default.
```rust
query.for_each_mut(|(mut transform, velocity)| {
    // ..logic
});
```

Reply:
Yeah, that's fair. It's mostly stylistic in this flavor of game though :p The Book shows the for loop style though, so I'll demonstrate `for_each` here.